### ISSUE TITLE
Sort out draw context vs visual properties

### ISSUE THREAD

**Issue by noahgibbs** [scarpe-team/scarpe#476]
There are a number of contextual properties (e.g. stroke, fill, strokewidth) that can be set globally, and then a drawable (e.g. an oval or arc) will pick up the current values when you create it, unless you override them.

The way we're doing that now is that we set the current values in a draw context, and then the drawable gets a copy of the draw context.

That's not terrible.

Unfortunately, drawables are supposed to make it possible to query the various individual styles (e.g. @oval.fill or @arc.strokewidth). Right now we're not seeing the ones from the draw context, plus we have to declare all those styles for each drawable for you can't call them. Not so good.

Plus you can override them. E.g.:

    oval(5, 10, 25, fill: orange)

I *think* the right answer is that instead of setting the draw context, the new widget should copy the properties from the draw context into its own styles, with the overrides doing what you'd expect. Then draw_context will never go outside of Lacci -- it will all turn into properties from the point of view of the display service.

**Comment by noahgibbs** [comment]
Come to think of it, in Drawable#initialize we should probably just see if the current drawable has any styles that can come from the draw context (stroke, fill, etc.) and just copy those in automatically. No need to declare them separately.

**Comment by noahgibbs** [comment]
So yeah, based on testing, draw context is set at drawable-creation time.

Draw context is *not* a single global. It's set per-slot, but inherits from the parent slot. Example:

```
Shoes.app do
  fill green
  stack(width: 100, height: 100) do
    background yellow
    # In Shoes3 this is still green. So parent-slot fill is used here.
    oval 10, 10, 40
  end

  flow do
    stack(width: 50, height: 50) do
      fill blue
    end
    stack(width: 50, height: 50) do
      # In Shoes3 this is green, not blue.
      # So it gets the parent draw context,
      # not the sibling draw context.
      oval 5, 5, 15
    end
  end
end
```

**Comment by noahgibbs** [comment]
Fixed by #513.
